src/test/ui/consts/const-eval/simd/insert_extract.rs

```diff
@@ -7,7 +7,8 @@
 
 #[repr(simd)] struct i8x1(i8);
```

Comment:
Can you check whether `#[repr(simd)] struct i8x1([i8; 1]);` won't end up with an immediate? Not that this type makes much sense, but probably good to be exhaustive.

Reply:
Good point, I have added that case to the test as well.